Core/EventHandlers.lua

```diff
@@ -1158,19 +1158,15 @@ function R:OnEvent(event, ...)
 			end
 		end
 
-		-- Handle opening Silver Strongbox & Gilded Chest (Shadowlands, Bastion nodes for Acrobatic Steward toy)
+		-- Handle opening Silver Strongbox & Gilded Chest (Bastiom, Shadowlands nodes for Acrobatic Steward (toy) and Gilded Wader (pet))
 		if
 			Rarity.isFishing and Rarity.isOpening and Rarity.lastNode and
 				(Rarity.lastNode == L["Silver Strongbox"] or Rarity.lastNode == L["Gilded Chest"])
 		 then
-			local names = {"Acrobatic Steward"}
-			if (Rarity.lastNode == L["Silver Strongbox"]) then
-				Rarity:Debug("Detected Opening on " .. L["Silver Strongbox"] .. " (method = SPECIAL)")
-			elseif (Rarity.lastNode == L["Gilded Chest"]) then
-				Rarity:Debug("Detected Opening on " .. L["Gilded Chest"] .. " (method = SPECIAL)")
-			end
+			local names = {"Acrobatic Steward", "Gilded Wader"}
+			Rarity:Debug("Detected Opening on " .. Rarity.lastNode .. " (method = SPECIAL)")
```

Comment:
Can this go wrong? I'm not sure if ``lastNode`` is always set properly, but if it isn't (and the value is ``nil``), this will cause a Lua error and stop execution of the script entirely.

Considering the dubious quality of the legacy code, I'd be very wary of these kinds of errors :worried:

Reply:
Well, I don't think that is a problem, as it already checks this in the outer if;
`if
			Rarity.isFishing and Rarity.isOpening and Rarity.lastNode and
				(Rarity.lastNode == L["Silver Strongbox"] or Rarity.lastNode == L["Gilded Chest"])
		 then`

if `Rarity.lastNode` isn't set with the right values or is `nil`, then that statement will be false, and thus it never gets this far

Reply:
Ah, indeed. I'm kind of losing track with all these conditions!